x-pack/plugin/security/qa/multi-cluster/src/javaRestTest/java/org/elasticsearch/xpack/remotecluster/RemoteClusterSecurityMlIT.java

```diff
@@ -0,0 +1,214 @@
+            putMlJobRequest2.setJsonEntity("""
+                {
+                  "analysis_config" : {
+                      "bucket_span": "1h",
+                      "detectors": [{"function":"sum","field_name":"responsetime","by_field_name":"airline"}]
+                  },
+                  "data_description" : {
+                      "format":"xcontent"
+                  },
+                  "datafeed_config": {
+                    "indexes":["my_remote_cluster:private-airline-data"]
+                  }
+                }
+                """);
+            final ObjectPath putMlJobResponse2 = assertOKAndCreateObjectPath(performRequestWithRemoteMlUser(putMlJobRequest2));
+            assertThat(putMlJobResponse2.evaluate("job_id"), equalTo("invalid"));
+            assertThat(putMlJobResponse2.evaluate("datafeed_config.job_id"), equalTo("invalid"));
+            assertThat(putMlJobResponse2.evaluate("datafeed_config.datafeed_id"), equalTo("invalid"));
+
+            assertOK(performRequestWithRemoteMlUser(new Request("POST", "/_ml/anomaly_detectors/invalid/_open")));
+
+            final ResponseException startDatafeedException = expectThrows(
+                ResponseException.class,
+                () -> performRequestWithRemoteMlUser(new Request("POST", "/_ml/datafeeds/invalid/_start"))
+            );
+            assertThat(startDatafeedException.getResponse().getStatusLine().getStatusCode(), equalTo(403));
+            assertThat(startDatafeedException.getMessage(), containsString("unauthorized for user [" + REMOTE_ML_USER + "]"));
+
+            final ResponseException previewDatafeedException = expectThrows(
+                ResponseException.class,
+                () -> performRequestWithRemoteMlUser(new Request("GET", "/_ml/datafeeds/invalid/_preview"))
+            );
+            assertThat(previewDatafeedException.getResponse().getStatusLine().getStatusCode(), equalTo(403));
+            assertThat(previewDatafeedException.getMessage(), containsString("unauthorized for user [" + REMOTE_ML_USER + "]"));
+        }
+    }
+
+    public void testDataframeAnalyticsNotSupportForRemoteIndices() {
+        final Request putDataframeAnalytics = new Request("PUT", "/_ml/data_frame/analytics/invalid");
+        putDataframeAnalytics.setJsonEntity("""
+            {
+              "source": {
+                "index": "my_remote_cluster:shared-airline-data"
+              },
+              "dest": {
+                "index": "data-frame-analytics-dest"
+              },
+              "analysis": {"outlier_detection":{}}
+            }
+            """);
+        final ResponseException e = expectThrows(ResponseException.class, () -> performRequestWithRemoteMlUser(putDataframeAnalytics));
+        assertThat(e.getResponse().getStatusLine().getStatusCode(), equalTo(400));
+        assertThat(e.getMessage(), containsString("remote source indices are not supported"));
```

Comment:
Explicitly testing that remote indices are currently not supported for data frame analytics. Hopefully this would give us a heads up in future if it becomes supported. A similar YAML test can be found at https://github.com/elastic/elasticsearch/blob/bbea1f15cf84d9e01521f18d7ec71a6df5b2a5a6/x-pack/plugin/src/yamlRestTest/resources/rest-api-spec/test/ml/data_frame_analytics_crud.yml#L435